app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
```

Comment:
According to [this line](https://github.com/pkarman/saml_idp/blob/logout-request-builder/lib/saml_idp/algorithmable.rb#L4), the gem will first look for a method called `raw_algorithm`, and if it's found, use its value, otherwise it will default to `SamlIdp.config.algorithm`. The way that code is written, it implies that there is a use case in which you want to be able to override the algorithm by defining a `raw_algorithm` method, and not by setting it in your config.

What is that use case?

According to this PR, we are using the algorithm defined in the config, so it seems like the only reason we are defining the `raw_algorithm` method in the controller is because the upstream code is written in a way that forces you to do that.

I would rather fix the upstream code so that it doesn't force you to define that method. If the method is not defined, it shouldn't crash with `undefined local variable or method 'raw_algorithm' for #<SamlIdpController:0x0056134ac0ccf0>`. I would expect it to gracefully fall back to `SamlIdp.config.algorithm`.

If the algorithm can always be read from the config, then this can be further simplified by removing references to `raw_algorithm`.